python/adelphi/bin/run-adelphi-tests.py

```diff
@@ -0,0 +1,77 @@
+# Run test suite for some set of Cassandra versions.
+#
+# We implement this as a front-end script because the tox/pytest/unittest chain
+# isn't great about iterating over a suite of test fixtures and re-running the
+# collected tests for each of them.  Rather than fight with the frameworks we
+# manage the fixture iteration manually in this script.  This also has the
+# nice side effect of moving a lot of C* checking/session code out of the test
+# suite, which in turn should allow us to write simpler tests.
+import configparser
+import os
+
+import click
+import tox
+
+import docker
+from tenacity import retry, stop_after_attempt, wait_fixed
+
+# Default C* versions to include in all integration tests
+DEFAULT_CASSANDRA_VERSIONS = ["2.1.22", "2.2.19", "3.0.23", "3.11.9", "4.0-rc1"]
+
+TOX_DEPENDENCIES = """pytest
+    subprocess32 ~= 3.5"""
+TOX_CONFIG = "tox.ini"
+
+
+@retry(stop=stop_after_attempt(3), wait=wait_fixed(2))
+def getContainer(client, version):
+    return client.containers.run(name="adelphi", remove=True, detach=True, ports={9042: 9042}, image="cassandra:{}".format(version))
+
+
+def writeToxIni(version):
+    config = configparser.ConfigParser()
+    config["tox"] = { "envlist": "py2, py3" }
+    envs = {"CASSANDRA_VERSION": version}
+    config["testenv"] = {"deps": TOX_DEPENDENCIES, \
+        "commands": "pytest {posargs}", \
+        "setenv": "CASSANDRA_VERSION = {}".format(version)}
+    with open(TOX_CONFIG, 'w') as configfile:
+        config.write(configfile)
+
+@click.command()
+@click.option('--cassandra', '-c', multiple=True, type=str)
+@click.option('--python', '-p', multiple=True, type=click.Choice(["py2","py3"], case_sensitive = False))
+@click.option("--pytest", "-t", type=str, help="Arguments to be passed to pytest")
```

Comment:
Click represents a significant improvement to usability here.  Using these args users can select (at run-time) any combination of the following:

* One or more Cassandra versions ("python bin/run-adelphi-tests.py -c 2.1.22 -c 2.2.19")
* Which Python version to use (if a specific one is required) ("python bin/run-adelphi-tests.py -p py2")
* Any args which should be passed directly to pytest (as executed by tox).  This allows users to leverage the full set of pytest functionality which is really pretty extensive.  (e.g. 'python bin/run-adelphi-tests.py -t "-k TestNb"' to find a specific test by keyword)

As expected these args can be mixed and matched as desired.

In some cases these params were passed via env vars, in some cases via tox args, and prolly a few other mechanisms I'm forgetting now.  Consolidating it all in a single place (with good --help support) should help quite a bit.